query-engine/connectors/query-connector/src/interface.rs

```diff
@@ -32,8 +32,9 @@ pub trait Connection: ConnectionLike {
 
 #[async_trait]
 pub trait Transaction: ConnectionLike {
-    async fn commit(&mut self) -> crate::Result<()>;
-    async fn rollback(&mut self) -> crate::Result<()>;
+    async fn begin(&mut self) -> crate::Result<()>;
```

Comment:
The name might be confusing because it might look like it begins the current transaction and needs to be called before anything else. Should we make it more clear that it begins a sub-transaction? Since the API models how savepoints work (i.e. you can't have concurrent/overlapping sub-transactions in a transaction), maybe exposing the `savepoint` terminology could be the right thing on this level of abstraction.

Reply:
Right now, the `begin` function will start a transaction or create a savepoint, depending on the current `depth` level. If a new transaction is started, and a transaction already exists with the same transaction ID, we call `begin` to increase the depth and start a sub-transaction.
What do you think of changing this so that we have a `savepoint` function that creates the savepoint, and keep the logic to begin a new transaction the same as it is in `main`? This would make it very clear that a sub-transaction is being started.